### DISCUSSION TITLE
Roadmap and Next.js development plans

### DISCUSSION THREAD

**Discussion by paqstd-dev** [vercel/next.js#65668] [App Router]
Almost 2.7k issues and this number is growing. Are there any plans to fix existing bugs? I wonder where all this is going, since with each new release of nextjs, using it in production becomes more and more problematic...

It is also interesting to know what plans there are for the development of the nextjs ecosystem.

**Comment by beinoriusju** [comment]
While product is free and talking is easy I won't complain, but have to agree with concerns expressed by OP. Since version 13 products is problematic to use and since 14 it's barely usable. Reading issues I see developers [struggling](https://github.com/vercel/next.js/issues/58263) making basic things work wasting hours of time.  Event ChatGPT stopped being helpful in navigating the changes generating Next 12 pages and oldie API routes. I'm no longer confident in choosing Next.js as web framework for new clients fearing getting stuck somewhere I should not get stuck.

One thing that would have saved everyone's time: deprecation of pages router with version 14 release. This would have signaled teams commitment to app router and freed developers mental energy from reading two documentations (router split). Why not have documentations version number related?

- Version 14 documentation
- Version 13 documentation
- Version 12 documentation
...

Feature releases should be planned accordingly to fall into version buckets and be be communicated once, ideally fully finished before next number is introduced.  Changes should be spaced out across numbers so people have benefit of semantic association.

DOUBT is the root of all evil.
